### ISSUE TITLE
Elementor not working with Cloudflare

### ISSUE THREAD

**Issue by AustinGil** [elementor/elementor#13647]
- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

## Description

I'm having problems with Elementor and Cloudflare lately. This is unrelated to the Rocket Loader issues, I know about that. But recently I’ve noticed that Elementor wont even open unless I completely disable the Cloudflare proxy. I've even tested Elementor's safe mode and the editor still does not load.

In Cloudflare I don’t have any of the performance optimizations enabled. No Rocket Loader, no Auto optimization, no Brotli. Still, the Elementor editor just will not load unless I totally disable the Cloudflare proxy.

I also notice that my WordPress backend in general is very slow. The only fix seems to be to disable Elementor, or Cloudflare. I've tried disabling all other plugins and tried switching themes but it did not help.

Occasionally I even get a timeout screen from Cloudflare, but this happens almost immediately which is very strange.

## Steps to reproduce

Install Elementor on a website with Cloudflare DNS proxy enabled.

## Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a default WordPress theme active.
- [X] I can reproduce this bug consistently using the steps above.

Sometimes it takes a while to test due to DNS propagation times. But I've tested several times and am pretty certain it's Elementor.

## Environment

<details>
<summary>System Info</summary>
```
https://pastebin.com/LMReCr7X
```
</details>

**Comment by parasshah195** [comment]
@AustinGil, all my sites use the combination of Cloudflare proxy DNS and Elementor (with E Pro), and I've never faced any issue with the editor loading (as long as I keep the Rocket Loader disable as advised).

Did you check the browser console to see if any JavaScript errors are coming up there?

Ideally, to troubleshoot, it's recommended to disable all the plugins keeping only Elementor active and see if the issue persists. It is possible that some other plugin on the site is conflicting when the combination of Cloudflare + Elementor is enabled.

You can use the "Health Check & Troubleshooting" plugin is temporarily disable all other plugins for you without actually disabling them for your site visitors - https://wordpress.org/plugins/health-check/

**Comment by shilo-ey** [comment]
@AustinGil Thanks for your report.

We have vast experience with Elementor sites based on CF DNS proxy with no issues reported.

As @parasshah195, can you please verify that any other theme/plugin/custom code is not interfering with your site load?

**Comment by AustinGil** [comment]
Thanks for the replies @parasshah195 and @shilo-ey. I'll give it another try with the health-check plugin enabled. Please give me a few days to make sure DNS propagation is working properly.

**Comment by AustinGil** [comment]
Yeah, this problem is still persisting. The good news for this thread is that I don't think it's actually related to Elementor.

![image](https://user-images.githubusercontent.com/4879075/105872857-baa19800-5faf-11eb-98be-068ec76a4de4.png)

I'm seeing other scripts not being loaded like WordPress' built in lodash and various others. They all return 522 errors, but it happens almost immediately so it really doesn't make a lot of sense :(

**Comment by EC1985** [comment]
Hi @AustinGil did you end up resolving your issue? We're encountering the same issue that you seem to have. Conflict between Cloudflare and (Elementor OR Wordpress). Been trying different solutions for months still haven't resolved it, then came across this thread

**Comment by AustinGil** [comment]
Hey @EC1985. I haven't seen the issue again in a while, but I'm not positive it's fixed. It was gone for a while, but lately my site has been slow on the admin side. Not seeing the 522 errors though.

I posted an update here https://wordpress.org/support/topic/elementor-not-working-with-cloudflare/?#post-14272946. Mostly came down to removing some of the page rules from Cloudflare for HTTPS redirects.

**Comment by EC1985** [comment]
Thanks Austin, I'll look that up and let you know if we find anything that
permanently solves the issues

**Comment by Tee-eN-Gee** [comment]
![image](https://user-images.githubusercontent.com/45270706/205747475-295152c3-df40-4f11-bd89-d39958ffede0.png)
Same issues

**Comment by rafaelgtrido** [comment]
Hello guy,

I had the same problem, in my case it was a CloudFlare Worker used to remove the utm parameters from the url. Diabling the worker, it starts working again. I still don't know how to make both work at the same time.
